File: AvgWaitTime.java
class Solution {
    public double averageWaitingTime(int[][] customers) {
        int nextIdleTime = 0; // Time when the chef will be next idle
        long netWaitTime = 0; // Total waiting time for all customers

        for (int[] customer : customers) {
            nextIdleTime = Math.max(customer[0], nextIdleTime) + customer[1];
            netWaitTime += nextIdleTime - customer[0];
        }

        return (double) netWaitTime / customers.length;
    }
}

File: AvgWaitTime.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        next_idle_time = 0  # Time when the chef will be next idle
        net_wait_time = 0  # Total waiting time for all customers

        for customer in customers:
            next_idle_time = max(customer[0], next_idle_time) + customer[1]
            net_wait_time += next_idle_time - customer[0]

        return net_wait_time / len(customers)
